lib/http/HttpClient_WinInet.cpp

```diff
@@ -339,16 +335,20 @@ class WinInetRequestWrapper
                         // must stay valid and writable until the next
                         // INTERNET_STATUS_REQUEST_COMPLETE callback comes
                         // (that's why those are member variables).
+                        LOG_TRACE("InternetReadFile() failed: ERROR_IO_PENDING. Waiting for INTERNET_STATUS_REQUEST_COMPLETE to be called again");
                         return;
                     }
                     LOG_WARN("InternetReadFile() failed: %d", dwError);
                     break;
                 }
 
                 m_bodyBuffer.insert(m_bodyBuffer.end(), m_buffer, m_buffer + m_bufferUsed);
-            } while (m_bufferUsed == sizeof(m_buffer));
+            } while (m_bufferUsed != 0);
         }
 
+        std::unique_ptr<SimpleHttpResponse> response(new SimpleHttpResponse(m_id));
+        response->m_body = m_bodyBuffer;
```

Comment:
On error, better to not copy the body buffer at all to prevent returning partial/corrupted body. So perhaps move this line to inside the L353 if block.

Reply:
good call, moved to if block below